src/main/frontend/handler/editor.cljs

```diff
@@ -781,8 +781,14 @@
     (save-block-if-changed! block new-content)))
 
 (defn set-marker
-  [{:block/keys [marker content] :as block} new-marker]
-  (let [new-content (->
+  [{:block/keys [marker content format] :as block} new-marker]
```

Comment:
With small changes like this, I think it gives us a good opportunity to add a few tests to demonstrate old behavior and new behavior. Since `save-block-if-changed!` makes testing this more difficult, we can use `with-redefs` to stub it out and just test the value of new-content. You're welcome to try this. I also don't mind doing a quick follow up PR with these tests if you give me some examples of marker, new-marker and new-content

Reply:
@logseq-cldwalker Thanks. I've added some tests for `set-maker` without using `with-redefs`. Does them work for you? I'll do some investigation on `with-redefs`.

Reply:
Thanks for adding some tests! I'll create a follow up PR that demonstrates with-redefs